### PR TITLE
Refactor inbox tab

### PR DESCRIPTION
**Pull request by mgcam**
Built on top of https://github.com/wtsi-npg/npg_langqc/pull/91

### CODE REVIEW

File: tests/test_wells4status_retrieval.py [hunk 1]
```diff
@@ -91,17 +91,82 @@ def test_query(qcdb_test_session, mlwhdb_test_session, load_data4well_retrieval)
...
-    )
-    with pytest.raises(Exception, match=r"Not implemented"):
-        factory.create()
+        qc_flow_status=status,
+    ).create()
+    assert isinstance(paged_wells, PacBioPagedWells)
+    assert paged_wells.page_size == 10
+    assert paged_wells.page_number == 1
+    assert paged_wells.qc_flow_status == status
+    assert paged_wells.total_number_of_items == 8
```
Comment (nerdstrike) [inline tests/test_wells4status_retrieval.py:110]: It's harder than I like to figure out how there are only 8 items in this inbox. Mutating fixtures and all that.

File: tests/test_wells4status_retrieval.py [hunk 2]
```diff
@@ -91,17 +91,82 @@ def test_query(qcdb_test_session, mlwhdb_test_session, load_data4well_retrieval)
...
+    _compare_dates(well.run_complete_time, well_fixture[3])
+    _compare_dates(well.well_start_time, well_fixture[4])
+    _compare_dates(well.well_complete_time, well_fixture[5])
+
+    well = paged_wells.wells[1]
+    assert isinstance(well, PacBioWell)
+    assert well.run_name == "TRACTION_RUN_12"
+    assert well.label == "A1"
+    assert well.qc_state is None
+    [well_fixture] = [
```
Comment (nerdstrike) [inline tests/test_wells4status_retrieval.py:163]: Sneaky. I didn't know python did this.

**Review by nerdstrike — COMMENTED** [review]
Observations and meditations. Will approve without further changes if you do not wish to spend any more time here..

File: tests/test_wells4status_retrieval.py (hunk 1 above)
Comment (mgcam) [reply tests/test_wells4status_retrieval.py:115]: Added comments, see https://github.com/wtsi-npg/npg_langqc/pull/93/commits/d0b604adcd9efcc1072c8b9339f09571a421909e

File: lang_qc/db/helper/wells.py [hunk 3]
```diff
@@ -179,3 +170,83 @@ def _add_tracking_info(self, wells: List[PacBioWell]):
...
+        # start date during the working day.
+        my_date = date.today() - timedelta(weeks=INBOX_LOOK_BACK_NUM_WEEKS)
+        start_datetime = datetime(my_date.year, my_date.month, my_date.day)
+
+        # TODO: fall back to run_complete when well_complete is undefined
+
+        query = (
+            select(PacBioRunWellMetrics)
+            .where(PacBioRunWellMetrics.well_status == "Complete")
+            .where(PacBioRunWellMetrics.run_complete > start_datetime)
```
[the remark this answers is not in the document]
Comment (mgcam) [reply lang_qc/db/helper/wells.py]: Variable renamed, see https://github.com/wtsi-npg/npg_langqc/pull/93/commits/76f7a0fba8a1ea18fa20075d98033f7ab10378fe

File: lang_qc/db/helper/wells.py [hunk 4]
```diff
@@ -179,3 +170,83 @@ def _add_tracking_info(self, wells: List[PacBioWell]):
...
+            .where(PacBioRunWellMetrics.polymerase_num_reads.is_not(None))
+            .where(
+                or_(
+                    and_(
+                        PacBioRunWellMetrics.ccs_execution_mode.in_(
```
[the remark this answers is not in the document]
Comment (mgcam) [reply lang_qc/db/helper/wells.py:204]: We has `is not None` there previously, which did not produce any error, but did not add the condition to the query. Having discovered this, I went for this explicit way of dealing with NULLs

File: lang_qc/db/helper/wells.py [hunk 5]
```diff
@@ -179,3 +170,83 @@ def _add_tracking_info(self, wells: List[PacBioWell]):
...
+
+    def _recent_inbox_wells(self, recent_wells):
+
+        inbox_wells_indexes = []
+        for index, db_well in enumerate(recent_wells):
+            in_qc = WellQc(
+                session=self.qcdb_session,
+                run_name=db_well.pac_bio_run_name,
+                well_label=db_well.well_label,
+            ).current_qc_state()
```
[the remark this answers is not in the document]
Comment (mgcam) [reply lang_qc/db/helper/wells.py:229]: I am reusing the class we have for setting QC states. What makes you think that it can go wrong?

**Review by nerdstrike — APPROVED** [review]

**Pull request merged**